File: CodeWarriors.IITDU/CodeWarriors.IITDU/Controllers/WishListController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using CodeWarriors.IITDU.Service;

namespace CodeWarriors.IITDU.Controllers
{
    [Authorize]
    public class WishListController : Controller
    {
        private readonly WishListService _wishListService;
        private readonly ProductService _productService;
        public WishListController(WishListService wishListService, ProductService productService)
        {
            _wishListService = wishListService;
            _productService = productService;
        }
        [HttpPost]
        public ActionResult AddToWishList(int productId)
        {
            var message = _wishListService.AddWishedProduct(productId, User.Identity.Name);
            return Json(message, JsonRequestBehavior.AllowGet);
        }
        [HttpPost]
        public ActionResult RemoveFromWishList(int wishId)
        {
            _wishListService.RemoveWishedProduct(wishId);
            return Json("Product removed from Wishlist successfully", JsonRequestBehavior.AllowGet);
        }

        public ActionResult GetWishList()
        {
            var wishList = _wishListService.GetWishList(User.Identity.Name);
            var ids = from item in wishList select item.ProductId;
            var products = _productService.GetAllProductsByProductId(ids.ToList());
            var wishItems = from wishedItem in wishList
                            from product in products where wishedItem.ProductId.Equals(product.ProductId)
                            select new { product = product, wishId = wishedItem.WishedProductId };
            //new { wishedItem.ProductId, wishedItem.WishedProductId, product.ProductName, product.Price }
            List<List<string>> images = wishItems.Select(wishedItem => _productService.GetProductImages(wishedItem.product.ProductId)).ToList();
            return Json(new { wishItems, images }, JsonRequestBehavior.AllowGet); ;
            //return Json(wishItems, JsonRequestBehavior.AllowGet);
        }

    }
}

File: CodeWarriors.IITDU/CodeWarriors.IITDU/App/Controllers/WishListController.js
app.controller("WishListController", function ($http, $scope) {
    init();
    function init() {
        $http.get("/Account/IsAuthenticated").success(function (response) {
            if (!response) {
                $location.path("home");
            }
        });
    }
    $scope.Message = "";
    $scope.Show = false;
    $scope.WishList = [];
    $http.get("/WishList/GetWishList").success(function (response) {
        //$scope.WishList = response;
        $scope.WishList = [];
        for (var i = 0; i < response.wishItems.length; i++) {
            $scope.WishList.push({ productInfo: response.wishItems[i].product, ImageUrl: response.images[i], wishId: response.wishItems[i].wishId });
        }
    });
    $scope.RemoveFromWishList = function (id) {
        var data = { wishId: parseInt(id) };
        $http.post("/WishList/RemoveFromWishList", data).success(function (response) {
            for (var i = $scope.WishList.length - 1; i >= 0; i--) {
                if ($scope.WishList[i].wishId == id) {
                    $scope.WishList.splice(i, 1);
                }
            }
            $scope.Show = true;
            $scope.Message = response;
        });
    }
    $scope.addToCart = function (id) {

        var cartItem = {
            ProductId: id,
            Quantity: 1
        };
        $http.post("/Cart/AddToCart/", cartItem).success(function (response, status) {
            if (status == 200) {
                $scope.Message = response;
                alert(response);
            }
            else $location.path("login");
        }).error(function () {
            $location.path("login");
        });
    }
})